src/Helper/SchemaFileHelper.php

```diff
@@ -75,4 +81,31 @@ public static function getAvroFiles(string $directory): array
 
         return $files;
     }
+
+    /**
+     * @param array $schema
+     * @return bool
+     */
+    public static function hasDocCommentsOnAllFields(array $schema): bool
+    {
+        $fields = $schema[self::FIELDS_FIELD_KEY] ?? null;
+
+        if (!is_array($fields) || count($fields) === 0) {
+            return true;
+        }
+
+        foreach ($fields as $field) {
+            $doc = $field[self::DOC_FIELD_KEY] ?? null;
+
+            if (!is_string($doc) || trim($doc) === '') {
+                return false;
+            }
+
+            if (false === self::hasDocCommentsOnAllFields($field)) {
```

Comment:
```suggestion
            if (false === self::hasDocCommentsOnAllFields($field['type'])) {
```
so if you are trying to check recursively in the schema folder, i think we need this
if we are only doing this check on the templates, then we don't need to check recursively.
I actually think it would be better to check the template directory, it is easier to follow, where you need to fix something , wdyt?

Reply:
Yeah, we will check templates, it is easier